photon-core/src/main/java/org/photonvision/vision/pipeline/ArucoPipeline.java

```diff
@@ -125,4 +165,18 @@ protected CVPipelineResult process(Frame frame, ArucoPipelineSettings settings)
 
         return new CVPipelineResult(sumPipeNanosElapsed, fps, targetList, frame);
     }
+
+    private Mat drawThresholdFrame(Mat greyMat, int windowSize, double constant) {
+        Mat thresh = new Mat();
+        if (windowSize % 2 == 0) windowSize++;
+        Imgproc.adaptiveThreshold(
+                greyMat,
+                thresh,
+                255,
+                Imgproc.ADAPTIVE_THRESH_MEAN_C,
+                Imgproc.THRESH_BINARY_INV,
+                windowSize,
+                constant);
+        return thresh;
```

Comment:
Why return a new mat, if we're instantly going to turn around and copy it? Have users pass in an input and output mat instead, and we can avoid a copy.